src/roster/views.py

```diff
@@ -1468,7 +1470,91 @@ def remove_social_account(request, *args, **kwargs):
+            customizations = Customization.objects.filter(book=book, periods=period)
+            comp_checks = ComprehensionCheckResponse.objects.filter(user=clusiveStudent, book=book)
+            custom_responses = ComprehensionCheckResponse.get_class_details_custom_responses(book=book, period=period)
+            custom_response = custom_responses.filter(user=clusiveStudent)
+            category_names = []
+            for category in book.reading_level_categories:
+                category_names.append(category.tag_name)
+
+            book_details.append({
+                'book_id': one_book['book_id'],
+                'title': one_book['title'],
+                'author': book.author,
+                'hours': round(one_book['hours'], 1),
+                'last_view': one_book['last_view'],
+                'view_count': one_book['view_count'],
+                'words_looked_up': ', '.join(json.loads(one_book['words_looked_up'])) if one_book['words_looked_up'] else None,
+                'first_version': one_book['first_version'],
+                'last_version': one_book['last_version'],
+                'custom_question': '(' + customizations[0].question + ')' if customizations else None,
+                'custom_response': custom_response[0].custom_response if custom_response else None,
+                'learning': comp_checks[0].get_answer if comp_checks else None,
+                'reading_level': ', '.join(category_names),
+                'is_assigned': one_book['is_assigned'],
+                'version_switched': True if one_book['first_version'] and one_book['first_version'] != one_book['last_version'] else False
+            })
+    return book_details    
+
+class ReadingDetailsPanelView(TemplateView):
+    """Shows just the reading details panel, for AJAX updates"""
+    template_name = 'roster/partial/student_details_reading_details.html'
+
+    def get(self, request, *args, **kwargs):
+        paginate_by = 10
+        paginate_orphans = 2
+
+        self.clusive_user = request.clusive_user
+        self.current_period = request.clusive_user.current_period
+        self.username = kwargs.get('username')
+        self.days = kwargs.get('days') if 'days' in kwargs else self.clusive_user.student_activity_days
+        self.page_num = kwargs.get('page_num') if 'page_num' in kwargs else 1
+        
+        if 'sort' in kwargs:
+            self.sort = kwargs.get('sort')
+            logger.debug('Setting reading details sort = %s', self.sort)
+            self.clusive_user.reading_details_sort = self.sort
+            self.clusive_user.save()
+        else:
+            self.sort = self.clusive_user.reading_details_sort
+
+        self.paginator = {}
+        self.page_obj = {}
+        try:
+            clusive_student = ClusiveUser.objects.get(
+                user__username=self.username,
+                periods__in=[self.current_period],
+                role__in=[Roles.STUDENT, Roles.GUEST]
```

Comment:
Guest not needed

Reply:
Fixed in base CSL-2046 student detail branch (PR #343) and merged herein.